index.js

```diff
@@ -8,12 +8,22 @@ module.exports = function(content) {
 	this.cacheable && this.cacheable();
 	if(!this.emitFile) throw new Error("emitFile is required from module system");
 	var query = loaderUtils.parseQuery(this.query);
+	var configKey = query.config || "fileLoader";
+	var config = this.options[configKey] || {};
```

Comment:
The only thing I question here, why publicPath option cannot be defined in the query? I think this is a webpack convention. 

By doing this, one could define publicPath option as a string in different loader, which is very convenient.